finality-aleph/src/party.rs

```diff
@@ -96,38 +95,30 @@ where
     let (ordered_batch_tx, ordered_batch_rx) = mpsc::unbounded();
     let (exit_tx, exit_rx) = futures::channel::oneshot::channel();
 
+    let keybox = KeyBox {
+        auth_keystore,
+        authorities: session.authorities.clone(),
+        id: node_id,
+    };
     let session_id = session.session_id as u64;
     let session_network =
-        session_manager.start_session(SessionId(session_id), session.authorities.clone());
+        RushNetwork::<B>::new(session_manager.start_session(SessionId(session_id), keybox.clone()));
 
+    let consensus_config =
+        default_aleph_config(session.authorities.len().into(), node_id, session_id);
```

Comment:
Oh goodness, no. There is some proper misdesign here, the config contains `session_id`, which should change each session. In the future also the node indes might change between sessions, as well as potentially the node count. Anyway, there pretty clearly is a part of the config that we want to keep the same between sessions and part that we want to change. Right now this is not expressed in any way in the code and requires a bigger refactor to make it even remotely sane – potentially needing even changes in AlephBFT (instead of `default_config` separate the delay config out and have a function for creating it – `default_delay_config`; perhaps also additional changes that I haven't though of yet).

I created a task for that but I definitely don't want to do this in this PR.